src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -296,7 +296,41 @@ module.exports = function<T, P, I, TI, C>(
     if (!coroutine) {
       throw new Error('Should be resolved by now');
     }
-    reconcileChildren(current, workInProgress, coroutine.children);
+
+    const nextChildren = coroutine.children;
+    const priorityLevel = workInProgress.pendingWorkPriority;
+    // At this point any memoization is no longer valid since we'll have changed
+    // the children.
+    workInProgress.memoizedProps = null;
+    if (!current) {
+      workInProgress.stateNode = mountChildFibersInPlace(
+        workInProgress,
+        workInProgress.stateNode,
+        nextChildren,
+        priorityLevel
+      );
+    } else if (current.child === workInProgress.child) {
+      clearDeletions(workInProgress);
+
+      workInProgress.stateNode = reconcileChildFibers(
+        workInProgress,
+        workInProgress.stateNode,
+        nextChildren,
+        priorityLevel
+      );
+
+      transferDeletions(workInProgress);
+    } else {
+      workInProgress.stateNode = reconcileChildFibersInPlace(
+        workInProgress,
+        workInProgress.stateNode,
+        nextChildren,
+        priorityLevel
+      );
+
+      transferDeletions(workInProgress);
+    }
+    // markChildAsProgressed(current, workInProgress, priorityLevel);
```

Comment:
This is a bit of a WIP still. It might be correctness related. I'm not even sure.